examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java

```diff
@@ -258,6 +263,108 @@ void createOtherGroup () {
+		editor.horizontalAlignment = SWT.LEFT;
+		editor.grabHorizontal = true;
+		// Add double click listener to the tree
+		tree1.addListener(SWT.MouseDoubleClick, new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				if (editableButton.getSelection()) {
+					Point point = new Point(event.x, event.y);
+					TreeItem item = tree1.getItem(point);
+					if (item != null) {
+						// Get the item text
+						final String oldText = item.getText();
+
+						// Create a text field to edit the item text
+						final Text text = new Text(tree1, SWT.NONE);
+						text.setText(oldText);
+						text.selectAll();
+						text.setFocus();
+
+						// Add a focus out listener to commit changes on focus lost
+						text.addListener(SWT.FocusOut, new Listener() {
+							@Override
+							public void handleEvent(Event e) {
+								item.setText(text.getText());
+								text.dispose(); // Dispose the text field after editing
+							}
+						});
+
+						// Add a key listener to commit changes on Enter key pressed
+						text.addListener(SWT.KeyDown, new Listener() {
+							@Override
+							public void handleEvent(Event e) {
+								if (e.keyCode == SWT.CR || e.keyCode == SWT.KEYPAD_CR) {
+									item.setText(text.getText());
+									text.dispose(); // Dispose the text field after editing
+								}
+							}
+						});
+
+						// Edit the text field on double click
+						editor.setEditor(text, item);
+					}
+				}
+			}
+		});
+
+		// Tree 2
+		final TreeEditor editor2 = new TreeEditor(tree2);
+		editor2.horizontalAlignment = SWT.LEFT;
+		editor2.grabHorizontal = true;
+		// Add double click listener to the tree
+		tree2.addListener(SWT.MouseDoubleClick, new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				if (editableButton.getSelection()) {
+					Point point = new Point(event.x, event.y);
+					TreeItem item = tree2.getItem(point);
+					if (item != null) {
+						// Get the item text
+						final String oldText = item.getText();
+
+						// Create a text field to edit the item text
+						final Text text = new Text(tree2, SWT.NONE);
+						text.setText(oldText);
+						text.selectAll();
+						text.setFocus();
+
+						// Add a focus out listener to commit changes on focus lost
+						text.addListener(SWT.FocusOut, new Listener() {
+							@Override
+							public void handleEvent(Event e) {
+								item.setText(text.getText());
+								text.dispose(); // Dispose the text field after editing
+							}
+						});
+
+						// Add a key listener to commit changes on Enter key pressed
+						text.addListener(SWT.KeyDown, new Listener() {
+							@Override
+							public void handleEvent(Event e) {
+								if (e.keyCode == SWT.CR || e.keyCode == SWT.KEYPAD_CR) {
+									item.setText(text.getText());
+									text.dispose(); // Dispose the text field after editing
+								}
+							}
+						});
+
+						// Edit the text field on double click
+						editor.setEditor(text, item);
+					}
```

Comment:
This looks like the exact same code as in the handler for other tree. Can it be moved to a separate method that is reused in both cases?